### PR TITLE
Release/snowplow normalize/0.1.0

### PR DESCRIPTION
**Pull request by rlh1994**
## Description & motivation
Latest commit updates docs and release numbers, update links for move into `snowplow` org, and adds values into the changelog for the release workflow to (hopefully) pull from. Latest commit also adds a different second event in the example to attempt to make it a bit better. 

Do not merge until repo has been moved to `snowplow` org and we have laucnh plan.

Overall release:
- Provides first full working version of packge, with python script split to functions and core run(there's maybe some more python design improvments that could be made in the future such as better use of classes or`__main__` but for now it's good enough to deploy).
- Provides tests on all python functions and core script output
- Provides 2 working macros for normalized events and user tables
- Documentation for script and macros, with more provided in docs PR
- Repo templates, labels, and workflows to match new repo design

## Checklist
- [x] I have verified that these changes work locally
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have raised a [documentation](https://github.com/snowplow/documentation) PR if applicable (https://github.com/snowplow/documentation/pull/150)

## Release Only Checklist
- [x] I have updated the version number in all relevant places
- [x] I have update the Github pages `dbt docs` 
- [x] I have updated the CHANGELOG.md

### CODE REVIEW

**Review by agnessnowplow — COMMENTED** [review]
Very impressive work!! I could see you put a lot of thought into it, neat comments where needed and clean code that is easy to follow and understand. The python is specifically 'advanced' with all the list comprehensions, function annotations etc. was a nice refresher + learning exercise for me to read through! I'll still have a look at the tests and do some further debugging once I have time, but all the rest seems to be working fine (apart from the macro call, but that might be just me not using as it was intended to (?). 

Just for my understanding, is filling out the config file a matter of a literal copy paste from their tracking files as well as defining the upper config variables or they would have to do some cosmetics, paste only parts of the code? As I did not have a source it was very fiddly for me so my initial thought was that it would be the easiest to just query the latest events table and get the latest contexts, sdes, and column types directly from there and have that as a base instead, but if they already have it there that's of course easy (in which case my bug I highlighted below might not even occur)

File: utils/snowplow_normalize_model_gen.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+{{%- set flat_cols = {flat_col or []} -%}}
+{{%- set sde_col = "{sde_col or ""}" -%}}
+{{%- set sde_keys = {sde_keys or []} -%}}
+{{%- set sde_types = {sde_types or []} -%}}
+{{%- set context_cols = {context_cols or []} -%}}
+{{%- set context_keys = {context_keys or []} -%}}
+{{%- set context_types = {context_types or []} -%}}
+{{%- set context_alias = {context_alias or []} -%}}
+
+{{{{ normalize_events(
```
[the remark this answers is not in the document]
Comment (rlh1994) [reply utils/snowplow_normalize_model_gen.py:199]: Ah that's 100% right, I've been testing it all in the package itself so that's a good catch!

File: utils/functions/snowplow_model_gen_funcs.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        else:
+            headers = {'apikey': api_key}
+            schema = requests.get(url, headers=headers).text
+        schema_cache[url] = schema
+    else:
+        verboseprint(f'Using cache for schema {url} ...')
+    schema = json.loads(schema)
+    return(schema)
+
+def validate_json(jsonData: dict, schema: dict = None, validate: bool = True, schemas_list: dict = None, repo_keys: dict = None) -> bool:
```
[the remark this answers is not in the document]
Comment (rlh1994) [reply utils/functions/snowplow_model_gen_funcs.py:127]: Sensible idea; I'll see how early in the code we can catch this as the models use some combination of event name/custom name plus version, so I don't genreate all the names until they are actually needed, but definitely a sensible thing to add!

**Comment by rlh1994** [comment]
> Just for my understanding, is filling out the config file a matter of a literal copy paste from their tracking files as well as defining the upper config variables or they would have to do some cosmetics, paste only parts of the code? As I did not have a source it was very fiddly for me so my initial thought was that it would be the easiest to just query the latest events table and get the latest contexts, sdes, and column types directly from there and have that as a base instead

@agnessnowplow It's not directly linked to their tracking files (unless they happen to record them in this specific way...) unfortunately, this is a specific file design for this package - I was hoping that the tracking catalogue on BDP would give users the URLs of the schemas so they would be easier to copy. Given you can have global contexts, wouldn't maybe want to do this for every event you are tracking, and in the next version I want to add support for multiple event types per model I think I'm okay with this being the manual work part of the package?

File: utils/functions/snowplow_model_gen_funcs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rlh1994) [reply utils/functions/snowplow_model_gen_funcs.py:127]: Okay I've added this and onto the add tests commit now - it should catch where the output file names are going to be duplicates of some others and raise a key error.

https://github.com/snowplow-incubator/dbt-snowplow-normalize/blob/5d289f72ce11474bcbd92d1c88c62a72a093d56c/utils/snowplow_normalize_model_gen.py#L71-L85

**Comment by rlh1994** [comment]
@agnessnowplow Added additional tests to catch the issues I had with the regex (table 4 on full run now includes that schema), and made the other chagnes you suggested and tests for them as well.

File: macros/users_table.sql [hunk 3]
```diff
@@ -68,7 +70,61 @@ where
 select
     * except (rn)
 from
-    users_ordered
+    users_ordering
+where
+    rn = 1
+{% endmacro %}
+
+{% macro databricks__users_table(user_cols, user_keys, user_types) %}
```
[the remark this answers is not in the document]
Comment (rlh1994) [reply macros/users_table.sql:78]: I think maybe that's the case normally but because I'm going directly from the schema content it isn't? Snowflake doesn't have any transformation on the schema field names, databricks and BQ are converted from camelCase to snake_case? (because snowflake just stores the raw json I think)?

[the remark this answers is not in the document]
Comment (agnessnowplow) [reply macros/users_table.sql:78]: You are right, the source is the schema itself and that is before any transformer/loader script making changes to it. It should be camelCase everywhere then. But the question is still the camelCase for Snowflake that needs changing: 
image

**Review by agnessnowplow — APPROVED** [review]
LGTM!

**Review by agnessnowplow — APPROVED** [review]
